plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -727,7 +727,26 @@ public void setTableOwner(String databaseName, String tableName, HivePrincipal p
     @Override
     public void commentColumn(String databaseName, String tableName, String columnName, Optional<String> comment)
     {
-        throw new TrinoException(NOT_SUPPORTED, "Column comment is not yet supported by Glue service");
+        Table oldTable = getExistingTable(databaseName, tableName);
+        List<Column> dataColumns = updateColumnComment(oldTable.getDataColumns(), columnName, comment);
+        List<Column> partitionColumns = updateColumnComment(oldTable.getPartitionColumns(), columnName, comment);
+        Table newTable = Table.builder(oldTable)
+                .setDataColumns(dataColumns)
+                .setPartitionColumns(partitionColumns)
+                .build();
+        replaceTable(databaseName, tableName, newTable, null);
+    }
+
+    private static List<Column> updateColumnComment(List<Column> columns, String columnName, Optional<String> comment)
+    {
+        return columns.stream()
+                .map(column -> {
+                    if (column.getName().equals(columnName)) {
+                        return new Column(column.getName(), column.getType(), comment, column.getProperties());
```

Comment:
- should the column name matching be case-insensitive?
- `updateColumnComment` should fail when no column matched, and when 2+ matched
   - and we should fail when data and partitioning column both found

Reply:
`TestHiveGlueMetastore` could potentially host the test

Reply:
> should the column name matching be case-insensitive?

To be extra safe we could but they don't need to be. Glue lowercases everything.

> updateColumnComment should fail when no column matched, and when 2+ matched
> and we should fail when data and partitioning column both found

Can do

Reply:
> To be extra safe we could but they don't need to be. Glue lowercases everything.
> 
> >

right. let's mention this wisdom in a code comment